src/components/fx/hover.js

```diff
@@ -402,10 +414,139 @@ function _hover(gd, evt, subplot, noHoverEvent) {
             hoverData.splice(0, closedataPreviousLength);
             distance = hoverData[0].distance;
         }
+
+        // Now look for the points to draw the spikelines
+        // Do it only if there is no hoverData
+        if(fullLayout._has('cartesian')) {
+            if(hoverData.length === 0) {
+                pointData.distance = spikedistance;
+                pointData.index = false;
+                var closestPoints = trace._module.hoverPoints(pointData, xval, yval, 'closest', fullLayout._hoverlayer);
+                if(closestPoints) {
+                    var tmpPoint;
+                    var closestVPoints = closestPoints.filter(function(point) {
+                        return point.xa.showspikes;
+                    });
+                    if(closestVPoints.length) {
+                        var closestVPt = closestVPoints[0];
+                        if(isNumeric(closestVPt.x0) && isNumeric(closestVPt.y0)) {
+                            tmpPoint = clearClosestPoint(closestVPt);
```

Comment:
This `clearClosestPoint` method is doing mostly initialisation as opposed to _clearing_ something that exists already.

Would be mind rename it either `initClosestPoint` or `fillClosestPoint`? This would make it more aligned with the rest of our codebase.

Reply:
You are right, I've renamed it to `fillClosestPoint`.